### PR TITLE
LCD 26543 client extension util spring mitigations

### PR DESCRIPTION
**Pull request by gamerson**
- LCD-26543 remove low-level infinite loop that can prevent cronjobs extensions from exiting
- LCD-26543 add a limited (5 min) retry loop to jwtDecoder bean only

### CODE REVIEW

**Comment by gamerson** [comment]
ci:forward

File: modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java [hunk 1]
```diff
@@ -122,14 +123,19 @@ public JwtDecoder jwtDecoder() throws Exception {
 			String clientId = _environment.getProperty(
 				externalReferenceCode + ".oauth2.user.agent.client.id");
 
-			if (clientId == null) {
+			long timeoutRemaining = TimeUnit.MINUTES.toMillis(5);
+
+			while ((clientId == null) && (timeoutRemaining > 0)) {
 				clientId = LiferayOAuth2Util.getClientId(
```
Comment (rotty3000) [inline modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java:129]: It seems the only thing using this timeout is `LiferayOAuth2Util.getClientId`? Could the timeout logic be moved inside the utility?

**Review by rotty3000 — CHANGES_REQUESTED** [review]

File: modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java (hunk 1 above)
Comment (gamerson) [reply modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java:129]: `LiferayOAuth2Util.getClientId` is called by cronjob clients (which caused original escalation).  So I wanted to move the while(true) out of the getClientId and move it to where it is actually needed (getting jwt / client ids for decoder).   We could make it even more higher level, but I feel this at least improves the status quo.

Comment (gamerson) [reply modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java:129]: Alternatively I could change the `.getClientId` method signature to take a flag to `retry=true` and move this look logic back there.

```
public static String getClientId(
		String externalReferenceCode, String lxcMainDomain,
		String lxcServerProtocol, boolean retry) {
```

Comment (gamerson) [reply modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java:129]: And then from OauthClientConfiguration (cronjobs) we would use `getClientId(false)` and from jwtDecoder bean we can use `getClientId(true)`

**Comment by gamerson** [comment]
ci:stop

File: modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java (hunk 1 above)
Comment (rotty3000) [reply modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java:129]: got it! my bad.

**Comment by gamerson** [comment]
ci:forward

**Pull request closed without merging**